util/windows/getopt/getopt.cpp

```diff
@@ -0,0 +1,698 @@
+/*
```

Comment:
I created this code few years ago, just reformatted for Posix standard.
on Windows platform there is no so big difference between C & C++ runtimes: there is same runtime library to support both languages. moreover, most of high-level interfaces like COM are implemented as C++ classes. NetworkDirect provider will be implemented mostly using C++ because it has COM based external interfaces